tls/tests/conftest.py

```diff
@@ -1,16 +1,80 @@
 # (C) Datadog, Inc. 2019-present
 # All rights reserved
 # Licensed under a 3-clause BSD style license (see LICENSE)
+import os
+
 import pytest
+from six import iteritems
 
+from datadog_checks.dev import TempDir, docker_run
 from datadog_checks.tls.utils import days_to_seconds
 
 from .utils import download_cert, temp_binary
 
+try:
+    from contextlib import ExitStack
+except ImportError:
+    from contextlib2 import ExitStack
+
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+
+CA_CERT = os.path.join(HERE, 'compose', 'ca.crt')
+
 
 @pytest.fixture(scope='session', autouse=True)
 def dd_environment():
-    yield {'server': 'https://www.google.com'}
+    with docker_run(
+        os.path.join(HERE, 'compose', 'docker-compose.yml'), build=True,
+    ):
+        e2e_metadata = {'docker_volumes': ['{}:{}'.format(CA_CERT, CA_CERT)]}
+        yield {'server': 'valid.mock'}, e2e_metadata
+
+
+@pytest.fixture(scope='session', autouse=True)
+def mock_dns():
+    import socket
+
+    _orig_getaddrinfo = socket.getaddrinfo
+
+    def patched_getaddrinfo(host, *args, **kwargs):
+        if host.endswith('.mock'):
+            # nginx doesn't support multiple tls versions from the same container
+            if 'tls-v1-1' in host:
+                port = 4444
+            elif 'tls-v1-2' in host:
+                port = 4445
+            elif 'tls-v1-3' in host:
+                port = 4446
+            else:
+                port = 4443
+
+            # See socket.getaddrinfo, just updating the hostname here.
+            return [(2, 1, 6, '', ('127.0.0.1', port))]
```

Comment:
What's `2, 1, 6`?

Reply:
Added a comment in the last commit. I just point to python docs, probably better than me trying to explain this.